pandas/core/arrays/boolean.py

```diff
@@ -286,6 +286,23 @@ def _from_sequence(cls, scalars, dtype=None, copy: bool = False):
         values, mask = coerce_to_array(scalars, copy=copy)
         return BooleanArray(values, mask)
 
+    @classmethod
+    def _from_sequence_of_strings(
+        cls, strings: List[str], dtype: Optional[str] = None, copy: bool = False
```

Comment:
the typings are not fully correct. `dtype` can be a dtype object, `string` an ndarray. 
Now since none of the other `_from_...` methods are typed, I wouldn't care about that too much on this PR